### PR TITLE
IDEMPIERE-6161:JasperReports encounters issues using encrypted functions due to incompatible dependencies

### PR DESCRIPTION
**Pull request by hieplq**
[IDEMPIERE-6161](https://idempiere.atlassian.net/browse/IDEMPIERE-6161)

**NOTE**: why don't update to 1.78.1?
The target platform encountered a resolution issue (Tycho was still able to build successfully)

The resolution issue can be addressed by disregarding the `includeDependencyScope='compile'` directive and including bcprov-jdk18on, bcutil-jdk18on, and bcpkix-jdk18on dependencies separately

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] I have checked my code and corrected any misspellings
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
### Tests
- [x] I have tested the direct scenario that my code is solving
- [ ] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

[IDEMPIERE-6161]: https://idempiere.atlassian.net/browse/IDEMPIERE-6161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hengsin** [comment]
need to change the sequence of the target platform.

File: org.idempiere.p2.targetplatform/org.idempiere.p2.targetplatform.target [hunk 1]
```diff
@@ -502,13 +502,13 @@ Export-Package:        *;-noimport:=true
...
       <artifactId>bcpg-jdk18on</artifactId>
-      <version>1.72.2</version>
+      <version>1.77</version>
       <type>jar</type>
 	</location>
 	<location includeDependencyScope="compile" includeSource="true" missingManifest="generate" type="Maven">
       <groupId>org.bouncycastle</groupId>
       <artifactId>bcmail-jdk18on</artifactId>
-      <version>1.72</version>
+      <version>1.77</version>
```
Comment (hengsin) [inline org.idempiere.p2.targetplatform/org.idempiere.p2.targetplatform.target:511]: please update the version in `maven.locations.xml` as well.

File: org.idempiere.test/src/org/idempiere/test/jasper/HandleSetupConfigurationPdfExport.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package org.idempiere.test.jasper;
```
Comment (hengsin) [inline org.idempiere.test/src/org/idempiere/test/jasper/HandleSetupConfigurationPdfExport.java:1]: please add license header

**Comment by CarlosRuiz-globalqss** [comment]
Confirmed the test case fails before and passes after the change.

File: org.idempiere.test/src/org/idempiere/test/jasper/HandleSetupConfigurationPdfExport.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	name = "IEventManager", bind = "bindEventManager", unbind="unbindEventManager", 
+	policy = ReferencePolicy.STATIC, cardinality =ReferenceCardinality.MANDATORY, service = IEventManager.class)
+)
+public class HandleSetupConfigurationPdfExport extends AbstractEventHandler {
+
+	// constant from JREventManage
+	public final static String JP_PDF_EXPORT_CONFIG_EVENT = "JP_PDF_EXPORT_CONFIG_EVENT";
+	public final static String JP_KEY_PDF_EXPORT_EXPORTER = "EXPORTER";
+	public final static String JP_KEY_PDF_EXPORT_CONFIG = "CONFIG";
+	public final static String JP_KEY_PROCESS_INFO = "PROCESS_INFO";
```
Comment (hengsin) [inline org.idempiere.test/src/org/idempiere/test/jasper/HandleSetupConfigurationPdfExport.java:25]: should use constant from `JREventManage` instead.

**Review by hengsin — CHANGES_REQUESTED** [review]
please address inline comments, thanks.

**Review by hengsin — APPROVED** [review]
looks good and tested by Carlos

**Pull request merged**